packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -811,25 +851,15 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
    *
    * @internal
    */
-  protected _createInstances(cluster: DatabaseClusterNew, props: DatabaseClusterProps): InstanceConfig {
+  protected _createInstances(props: DatabaseClusterProps): InstanceConfig {
     const instanceEndpoints: Endpoint[] = [];
     const instanceIdentifiers: string[] = [];
     const readers: IAuroraClusterInstance[] = [];
 
-    let monitoringRole = props.monitoringRole;
-    if (!props.monitoringRole && props.monitoringInterval && props.monitoringInterval.toSeconds()) {
-      monitoringRole = new Role(cluster, 'MonitoringRole', {
-        assumedBy: new ServicePrincipal('monitoring.rds.amazonaws.com'),
-        managedPolicies: [
-          ManagedPolicy.fromAwsManagedPolicyName('service-role/AmazonRDSEnhancedMonitoringRole'),
-        ],
-      });
-    }
-
     // need to create the writer first since writer is determined by what instance is first
     const writer = props.writer!.bind(this, this, {
-      monitoringInterval: props.monitoringInterval,
-      monitoringRole: monitoringRole,
+      monitoringInterval: props.enableClusterLevelEnhancedMonitoring ? undefined : props.monitoringInterval,
+      monitoringRole: props.enableClusterLevelEnhancedMonitoring ? undefined : this.monitoringRole,
```

Comment:
Would be nice to have a comment to explain that when `enableClusterLevelEnhancedMonitoring` is enabled, we already set interval and role at cluster level so no need to re-set it in instance level.

Reply:
I have added such explanations to all the sections where instance-level enhanced monitoring settings are applied.